swift/swift.py

```diff
@@ -16,12 +16,95 @@
 import importlib
 import importlib.util
 from contextlib import contextmanager
+from dataclasses import dataclass, asdict
+from typing import Any, Iterable, Mapping, Self
 
 from PyQt5.QtCore import QObject, pyqtSlot, Qt
 from PyQt5.QtWidgets import QApplication, QMainWindow, QDockWidget
 
 from swift.bus import Bus
 
+@dataclass
+class AppInfo:
+    """Information required to create an app.
+    
+    Fields:
+        module: Module name in which the app class resides.
+        class_: App class name.
+        path: System path for module importing.
+        show: Whether to show the app frames on creation.
+        pos: Position on the main window; refer to Qt.DockWidgetArea enum.
+          Should be one of "left", "right", "top", or "bottom", case-sensitive.
+          Otherwise, defaults to Qt.AllDockWidgetAreas.
+        bus: Buses which the app subscribes to.
+        args: Keyword argument dictionary of the app class constructor.
+          It must exclude name and parent arguments. Even if they exist, they will be ignored.
+          None for initializing the app with default values,
+          where only name and parent arguments will be passed.
+    """
+    module: str
+    class_: str
+    path: str = "."
+    show: bool = True
+    pos: str = ""
+    bus: Iterable[str] = ()
+    args: Mapping[str, Any] | None = None
+
+    @classmethod
+    def parse(cls, info: str) -> Self:
+        """Constructs an AppInfo object from a JSON string.
+        
+        Args:
+            info: A JSON string of a dictionary that contains the information of an app.
+              Its keys are field names of AppInfo and values are corresponding values.
+              Exceptionally for "class_" field, "class" is also accepted.
+              If both "class_" and "class" exist, then "class" is ignored.
+              If both does not exist, a KeyError("class_") is raised.
+        
+        Raises:
+            KeyError: When there is no mandatory fields in info.
+        """

```

Comment:
Like other functions with returning value, how about adding 'Return' explanation?

Reply:
I think the summary tells us enough information about the returning value.
Or should I change the summary to start with "Returns an App Info ..." instead of "Constructs an AppInfo ..."?

Reply:
I think that writing like "Returns an App Info ..." is better to understand!